### PR TITLE
added typescript documentation to plugins.md

### PR DESCRIPTION
**Pull request by mrcwbr**
as you mentioned in https://github.com/vuejs/test-utils/pull/1792 I created a custom plugin. if you want to use plugins in a typescript project you have to redefine the declaration of the `VueWrapper`. Therefore, I updated the docs

### CODE REVIEW

**Review by cexbrayat — APPROVED** [review]
LGTM thanks!

Can you rebase and resolve the conflicts so we can merge it?

File: src/baseWrapper.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export default abstract class BaseWrapper<ElementType extends Node>
     return this.wrapperElement
   }
 
-  constructor(element: ElementType) {
+  protected constructor(element: ElementType) {
```
Comment (lmiller1990) [inline src/baseWrapper.ts:44]: I am not sure this does anything - this class is already `abstract` so you cannot make an instance of it anyway. Might be good to revert an unnecessary change, unless I'm missing something?

**Comment by mrcwbr** [comment]
> LGTM thanks!
> 
> Can you rebase and resolve the conflicts so we can merge it?

done
